docs/zkapps/tutorials/09-recursion.mdx

```diff
@@ -56,86 +56,310 @@ You build recursive zkApps with [ZkProgram](/zkapps/o1js-reference/modules/Exper
+```
+
+These items are:
+
+- `Field`: The native number type in o1js. You can think of Field elements as unsigned integers. Field elements are the most basic type in o1js. All other o1js-compatible types are built on top of Field elements.
+- `ZkProgram`: The o1js general purpose API for creating zero knowledge proofs. A ZkProgram is similar to zkApp smart contracts but isn't tied to an on-chain account.
+
+### ZkProgram
+
+To create a ZkProgram, start with the `init()` method.
+
+- For each method, declare the inputs it will receive.
+- The first argument of a ZkProgram method is always the state of the ZkProgram, named `publicInput` since it is public.
+
+```typescript
+3 export const Add = ZkProgram({
+4  name: 'add-example',
+5  publicInput: Field,
+6
+7  methods: {
+8    init: {
+9      privateInputs: [],
+10
+11      method(state: Field) {
+12        state.assertEquals(Field(0));
+13      },
+14    },
+15  },
+16 });
+17
 ```
 
 Add another method that takes an existing proof, adds a new number to it, and produces a new proof:
 
 ```typescript
-    addNumber: {
-      privateInputs: [SelfProof, Field ],
-
-      method(newState: Field, earlierProof: SelfProof<Field>, numberToAdd: Field) {
-        earlierProof.verify();
-        newState.assertEquals(earlierProof.publicInput.add(numberToAdd));
-      },
-    },
+15
+16    addNumber: {
+17      privateInputs: [SelfProof, Field],
+18
+19      method(
+20        newState: Field,
+21        earlierProof: SelfProof<Field, void>,
+22        numberToAdd: Field
+23      ) {
+24        earlierProof.verify();
+25        newState.assertEquals(earlierProof.publicInput.add(numberToAdd));
+26      },
+27    },
+28  },
+29 });
 ```
 
 Use recursion to combine two proofs:
 
 ```typescript
-  add: {
-    privateInputs: [ SelfProof, SelfProof ],
-
-    method(
-      newState: Field,
-      earlierProof1: SelfProof<Field>,
-      earlierProof2: SelfProof<Field>,
-    ) {
-      earlierProof1.verify();
-      earlierProof2.verify();    	
-      newState.assertEquals(earlierProof1.publicInput.add(earlierProof2.publicInput));
-    },
-  },
+28
+29    add: {
+30      privateInputs: [SelfProof, SelfProof],
+31
+32      method(
+33        newState: Field,
+34        earlierProof1: SelfProof<Field, void>,
+35        earlierProof2: SelfProof<Field, void>
+36      ) {
+37        earlierProof1.verify();
+38        earlierProof2.verify();
+39        newState.assertEquals(
+40          earlierProof1.publicInput.add(earlierProof2.publicInput)
+41        );
+42      },
+43    },
+44  },
+45 });
 ```
 
-To use ZkProgram, compile it and then call methods on it:
+## Interact with a smart contract
+
+Next, write a script that interacts with your smart contract. As before, the complete [main.ts](https://github.com/o1-labs/docs2/blob/main/examples/zkapps/01-hello-world/src/main.ts) example file is provided. Follow these steps to build the `main.ts` file so you can interact with the smart contract.
```

Comment:
since Tutorial 4 was the "model" for these updates, it talks about interacting with the smart contract. We can omit the UI-based steps here